cartocrow/renderer/geometry_widget.cpp

```diff
@@ -680,6 +682,11 @@ void GeometryWidget::addPainting(std::shared_ptr<GeometryPainting> painting, con
 	updateLayerList();
 }
 
+void GeometryWidget::addPainting(const std::function<void(renderer::GeometryRenderer&)>& draw_function, const std::string& name) {
```

Comment:
I did not think of that.
That does mean you cannot do
```
renderer.addPainting([](GeometryRenderer& renderer) {
    renderer.draw(/* whatever you want to draw */);
});
```
because you need to still create a shared pointer, which I would like to avoid having to do.